Frameworks/CoreGraphics/CGContextCairo.mm

```diff
@@ -1977,3 +1985,66 @@
     cairo_path_destroy(caPath);
     return (CGPathRef)copyPath;
 }
+
+/**
+ * Helper method to render text with the given DWRITE_GLYPH_RUN.
+ * Today we create a IWICBitmap with the underlying XAML Writable Bitmap raw bytes and rasterize to it before copying it back to the
+ * original buffer.
+ * TODO::
+ *    1. D2D applies rotation transform clockwise but iOS does it anti-clockwise.
+ *    2. There is double copy happening here - first when the IWICBitmap is created and later the rasterized image is copied back
+ *       Implementing our own IWICBitmap interface should fix this issue.
+ *
+ * @parameter glyphRun DWRITE_GLYPH_RUN object to render
+ */
+void CGContextCairo::CGContextDrawGlyphRun(const DWRITE_GLYPH_RUN* glyphRun) {
+    ObtainLock();
+
+    CGContextStrokePath();
+
+    // Lock the data that is backing the layer.
+    BYTE* buffer = static_cast<BYTE*>(_imgDest->Backing()->LockImageData());
+    THROW_NS_IF_NULL(E_UNEXPECTED, buffer);
+    int stride = _imgDest->Backing()->BytesPerRow();
+    float width = _imgDest->Backing()->Width();
+    float height = _imgDest->Backing()->Height();
+
+    // Create a IWICBitmap object out of the memory that is backing the layer.
+    ComPtr<IWICImagingFactory> wicFactory;
+    THROW_IF_FAILED(CoCreateInstance(CLSID_WICImagingFactory, NULL, CLSCTX_INPROC_SERVER, __uuidof(IWICImagingFactory), &wicFactory));
+
+    ComPtr<IWICBitmap> wicBitmap;
+    THROW_IF_FAILED(
+        wicFactory->CreateBitmapFromMemory(width, height, GUID_WICPixelFormat32bppPBGRA, stride, height * stride, buffer, &wicBitmap));
+
+    // Create a ID2D1RenderTarget from the newly created IWICBitmap object.
+    ComPtr<ID2D1Factory> d2dFactory;
+    THROW_IF_FAILED(D2D1CreateFactory(D2D1_FACTORY_TYPE_SINGLE_THREADED, __uuidof(ID2D1Factory), &d2dFactory));
+    ComPtr<ID2D1RenderTarget> renderTarget;
+    THROW_IF_FAILED(d2dFactory->CreateWicBitmapRenderTarget(wicBitmap.Get(), D2D1::RenderTargetProperties(), &renderTarget));
+
+    // Set the brush color to the current values from the context.
+    D2D1::ColorF brushColor =
+        D2D1::ColorF(curState->curFillColor.r, curState->curFillColor.g, curState->curFillColor.b, curState->curFillColor.a);
+
+    renderTarget->BeginDraw();
+
+    // Apply the required transformations as set in the context.
+    CGAffineTransform transform = CGAffineTransformConcat(curState->curTransform, curState->curTextMatrix);
```

Comment:
>curState->curTransform [](start = 58, length = 22)

as discussed yesterday, should instead use a matrix with -b and -c in place of b and c